include/boost/beast/core/impl/buffers_adaptor.hpp

```diff
@@ -34,332 +35,162 @@ namespace beast {
-        mutable_buffers_type const&) = default;
-
-    const_iterator
-    begin() const;
-
-    const_iterator
-    end() const;
-
-private:
-    friend class buffers_adaptor;
-
-    mutable_buffers_type(
-            buffers_adaptor const& b)
-        : b_(&b)
+    struct iterator_category
+        : std::bidirectional_iterator_tag
     {
-    }
-};
-
-template<class MutableBufferSequence>
-class buffers_adaptor<MutableBufferSequence>::
-mutable_buffers_type::const_iterator
-{
-    iter_type it_{};
-    buffers_adaptor const* b_ = nullptr;
-
-public:
-    using value_type = net::mutable_buffer;
-    using pointer = value_type const*;
-    using reference = value_type;
+    };
+    using value_type = typename buffers_adaptor<MutableBufferSequence>::
+    template subrange<isMutable>::
+    value_type;
+    using reference = typename std::add_lvalue_reference<value_type>::type;
+    using pointer = typename std::add_pointer<value_type>::type;
     using difference_type = std::ptrdiff_t;
-    using iterator_category =
-        std::bidirectional_iterator_tag;
 
-    const_iterator() = default;
-    const_iterator(const_iterator const& other) = default;
-    const_iterator& operator=(const_iterator const& other) = default;
+    iterator(
+        typename buffers_adaptor<MutableBufferSequence>::
+        template subrange<isMutable> const *parent,
+        iter_type it);
 
-    bool
-    operator==(const_iterator const& other) const
-    {
-        return b_ == other.b_ && it_ == other.it_;
-    }
+    iterator();
 
-    bool
-    operator!=(const_iterator const& other) const
-    {
-        return !(*this == other);
-    }
+    value_type
+    operator*() const;
 
-    reference
-    operator*() const
-    {
-        value_type const b = *it_;
-        return value_type{b.data(),
-            it_ == std::prev(b_->end_) ?
-                b_->out_end_ : b.size()} +
-                    (it_ == b_->out_ ? b_->out_pos_ : 0);
-    }
+    iterator &
+    operator++();
 
-    pointer
-    operator->() const = delete;
+    iterator
+    operator++(int);
 
-    const_iterator&
-    operator++()
-    {
-        ++it_;
-        return *this;
-    }
+    iterator &
+    operator--();
 
-    const_iterator
-    operator++(int)
-    {
-        auto temp = *this;
-        ++(*this);
-        return temp;
-    }
+    iterator
+    operator--(int);
 
-    const_iterator&
-    operator--()
-    {
-        --it_;
-        return *this;
-    }
+    bool
+    operator==(iterator const &b) const;
 
-    const_iterator
-    operator--(int)
-    {
-        auto temp = *this;
-        --(*this);
-        return temp;
-    }
+    bool
+    operator!=(iterator const &b) const;
 
 private:
-    friend class mutable_buffers_type;
 
-    const_iterator(buffers_adaptor const& b,
-            iter_type iter)
-        : it_(iter)
-        , b_(&b)
-    {
-    }
+    buffers_adaptor<MutableBufferSequence>::
+    subrange<isMutable> const *parent_;
```

Comment:
continuation lines should be indented
```
    buffers_adaptor<MutableBufferSequence>::
        subrange<isMutable> const *parent_;
```
I'm not a fan of "parent" but it isn't a showstopper. I would have just used `b_`.

Reply:
I think `b_` is un-necessarily cryptic - I speak as someone who used to be quite happy being limted to 6 characters for label names in 6809 assembler. Even then we'd try to spll wt mnngfl nms.